coaster-blas/src/frameworks/cuda/helper.rs

```diff
@@ -106,6 +106,7 @@ macro_rules! iblas_copy_for_cuda {
             x: &SharedTensor<$t>,
             y: &mut SharedTensor<$t>,
         ) -> Result<(), ::coaster::error::Error> {
+            assert_eq!(x.desc().size(), y.desc().size());
```

Comment:
Technically it'd be fine if y was larger, but I am not sure this is a sane case given the context of shared tensors.

Reply:
+1. It would make sense if we can copy with an offset, but since we can't, I think we should enforce the sizes to be equal.